Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -953,6 +955,25 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         completionHandler?(nil)
     }
 
+    public func handleShutDown(completionHandler: ((Data?) -> Void)? = nil) {
+        Task {
+            let managers = try await NETunnelProviderManager.loadAllFromPreferences()
+
+            guard let manager = managers.first else {
+                return
+            }
+
+            manager.isOnDemandEnabled = false
+            try await manager.saveToPreferences()
+            try await manager.loadFromPreferences()
+
+            let error = NSError(domain: "com.duckduckgo.vpn", code: 0)
```

Comment:
I expect this error value will get picked up by the metrics as a legitimate error when shutting down the VPN, so need to test it a bit more.